xlators/cluster/afr/src/afr-open.c

```diff
@@ -311,9 +325,47 @@ afr_fix_open(fd_t *fd, xlator_t *this)
     ret = loc_path(&local->loc, NULL);
     if (ret < 0)
         goto out;
+    local->cont.lk.getlk_rsp = GF_CALLOC(sizeof(*local->cont.lk.getlk_rsp),
```

Comment:
This data structure is used and allocated in other parts of the code. Does it get allocated twice if that lock healing part is enabled? It would be a good idea to alloc and free in the init and free functions of afr_local_t if both don't  use it at the same time. If both can use it (parallel FOPS etc), then it is better to use another structure.

Reply:
Added another structure since it looks like there can be parallel FOPs using the same data structure if both strict locks & lock healing are enabled.